plugins/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -104,56 +103,114 @@
   hardware:
     description:
     - Manage VM's hardware parameters. VM needs to be shut down to reconfigure these parameters.
-    - 'Valid parameters are:'
-    - ' - C(num_cpus) (integer): Number of CPUs.'
-    - ' - C(num_cpu_cores_per_socket) (integer): Number of Cores Per Socket. C(num_cpus) has to be a multiple of C(num_cpu_cores_per_socket).'
-    - ' - C(memory_mb) (integer): Amount of memory in MB.'
     type: dict
+    suboptions:
+      num_cpus:
+        description: Number of CPUs.
+        type: int
+      num_cpu_cores_per_socket:
+        description: Number of Cores Per Socket. C(num_cpus) has to be a multiple of C(num_cpu_cores_per_socket).
+        type: int
+      memory_mb:
+        description: Amount of memory in MB.
+        type: int
   disks:
     description:
     - A list of disks to add to VM.
     - All parameters are case sensitive.
     - Removing or detaching existing disks of VM is not supported.
-    - 'Required parameters per entry:'
-    - ' - C(size_[tb,gb,mb,kb,b]) (integer): Disk storage size in specified unit. VM needs to be shut down to reconfigure this parameter.'
-    - 'Optional parameters per entry:'
-    - ' - C(name) (string): Disk name. You can also use C(name_label) as an alias.'
-    - ' - C(name_desc) (string): Disk description.'
-    - ' - C(sr) (string): Storage Repository to create disk on. If not specified, will use default SR. Cannot be used for moving disk to other SR.'
-    - ' - C(sr_uuid) (string): UUID of a SR to create disk on. Use if SR name is not unique.'
+    - Entries are required to have either a I(size) or one of I(size_[tb,gb,mb,kb,b]) parameters.
+    - VM needs to be shut down to reconfigure this parameter.
```

Comment:
This should read:

```- VM needs to be shut down to reconfigure **disk size**.```

If you say `this parameter` it is misleading. It could mean that reconfiguring anything in `disks:` requires VM to be shut down which is not true. Adding new disks and renaming existing ones can be done online. Resizing is the problem.